src/ast/semantic_analyser.cpp

```diff
@@ -991,18 +1009,21 @@ void SemanticAnalyser::visit(Cast &cast)
 {
   cast.expr->accept(*this);
 
-  const std::map<std::string, std::tuple<size_t, bool>> intcasts = {
-      {"uint8", std::tuple<size_t, bool>{1, false}},
-      {"int8", std::tuple<size_t, bool>{1, true}},
-      {"uint16", std::tuple<size_t, bool>{2, false}},
-      {"int16", std::tuple<size_t, bool>{2, true}},
-      {"uint32", std::tuple<size_t, bool>{4, false}},
-      {"int32", std::tuple<size_t, bool>{4, true}},
-      {"uint64", std::tuple<size_t, bool>{8, false}},
-      {"int64", std::tuple<size_t, bool>{8, true}},
-  };
-
   auto k_v = intcasts.find(cast.cast_type);
+  int cast_size;
```

Comment:
are you changing the behaviour here or just refactoring?

Reply:
Yes, it changes the behavior.
Here, the function checks `cast.is_pointer` at first. Previously [the check whether the cast type was int was first](https://github.com/iovisor/bpftrace/pull/942/files/1bc0eafe3316e3e8ee4dada07f563c975ba6569a#diff-286594496696c8553c9f12fcabfb5f38L1008) and therefore a pointer of int was not considered.